File: server/battle/src/main/java/info/chenliang/moba/message/PlayerToBattleDispatcher.java
package info.chenliang.moba.message;
import java.io.*;
import info.chenliang.talky.*;
public class PlayerToBattleDispatcher {
public static void dispatch(DataInputStream dis, PlayerToBattleProxy proxy) throws Exception{
int fid = (int)(dis.read()&0xFFFF);
switch(fid)
{
case 0:
{
playerPing(dis, proxy);
}
break;
case 1:
{
playerPong(dis, proxy);
}
break;
case 2:
{
play(dis, proxy);
}
break;
case 3:
{
moveTo(dis, proxy);
}
break;
}
}
protected static void playerPing(DataInputStream dis, PlayerToBattleProxy proxy) throws Exception{
proxy.playerPing();
}
protected static void playerPong(DataInputStream dis, PlayerToBattleProxy proxy) throws Exception{
proxy.playerPong();
}
protected static void play(DataInputStream dis, PlayerToBattleProxy proxy) throws Exception{
proxy.play();
}
protected static void moveTo(DataInputStream dis, PlayerToBattleProxy proxy) throws Exception{
float x;
float z;
x = dis.readFloat();
z = dis.readFloat();
proxy.moveTo(x,z);
}
}

File: client/Assets/Scripts/Message/info/chenliang/moba/message/PlayerToBattleDispatcher.cs
namespace info.chenliang.moba.message{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public class PlayerToBattleDispatcher {
public static void Dispatch(MemoryEndianBinaryReader reader, PlayerToBattleProxy proxy) {
int fid = reader.ReadByte();
switch(fid)
{
case 0:
{
playerPing(reader, proxy);
}
break;
case 1:
{
playerPong(reader, proxy);
}
break;
case 2:
{
play(reader, proxy);
}
break;
case 3:
{
moveTo(reader, proxy);
}
break;
}
}
protected static void playerPing(MemoryEndianBinaryReader reader, PlayerToBattleProxy proxy){
proxy.playerPing();
}
protected static void playerPong(MemoryEndianBinaryReader reader, PlayerToBattleProxy proxy){
proxy.playerPong();
}
protected static void play(MemoryEndianBinaryReader reader, PlayerToBattleProxy proxy){
proxy.play();
}
protected static void moveTo(MemoryEndianBinaryReader reader, PlayerToBattleProxy proxy){
Single x;
Single z;
x = reader.ReadSingle();
z = reader.ReadSingle();
proxy.moveTo(x,z);
}
}
}
